internal/compiler/generator/rust.rs

```diff
@@ -1277,41 +1282,60 @@ fn generate_item_tree(
         quote!(&self_rc)
     };
 
-    let (create_window_adapter, init_window, new_result, new_end) = if let Some(parent_ctx) =
-        parent_ctx
-    {
-        (
-            None,
-            None,
-            quote!(vtable::VRc<slint::private_unstable_api::re_exports::ComponentVTable, Self>),
-            if parent_ctx.repeater_index.is_some() {
-                // Repeaters run their user_init() code from RepeatedComponent::init() after update() initialized model_data/index.
-                quote!(self_rc)
-            } else {
-                quote! {
-                    Self::user_init(slint::private_unstable_api::re_exports::VRc::map(self_rc.clone(), |x| x));
-                    self_rc
-                }
-            },
-        )
-    } else {
-        (
-            Some(
-                quote!(let window_adapter = slint::private_unstable_api::create_window_adapter()?;),
-            ),
-            Some(quote! {
-                _self.window_adapter.set(window_adapter);
-                slint::private_unstable_api::re_exports::WindowInner::from_pub(_self.window_adapter.get().unwrap().window()).set_component(&VRc::into_dyn(self_rc.clone()));
-            }),
-            quote!(
-                core::result::Result<
-                    vtable::VRc<slint::private_unstable_api::re_exports::ComponentVTable, Self>,
-                    slint::PlatformError,
-                >
-            ),
-            quote!(core::result::Result::Ok(self_rc)),
-        )
-    };
+    let (window_adapter_accessor, window_adapter_ref, maybe_window_adapter, new_result, new_end) =
```

Comment:
this whole thing is getting hard to read.
But i'm not sure how to simplify it.
Maybe splitting in many ifs?
Or having less quote!() with some duplication between?

Reply:
Good that you bring it up. I have an idea how to make it more readable.